### PR TITLE
Mock request when testing StorageClient

### LINKED ISSUE #777: Fix tests in plot_storage_api that have been commented out

### PR DESCRIPTION
**Pull request by lars-petter-hauge**
**Issue**
Resolves #777

**Approach**
What we want to test is that the `StorageClient` acts properly, and we should not need to care about the end-point. I would then rather mock the return values from requests.

However in these days the API's can be changing quite fast, and I think some of the purpose of these tests are to verify the integrity all the way from the new storage solution and up to the plotter.

Therefore my approach was to use the `FlaskWrapper` and reuse the in-memory database that we have. Hence I mocked the entire `request` import in `storage_client.py` to be the `test_client` that Flask provides.

My only issue now though is that the `test_client` from Flask returns a `Flask.Response` object, and not a `request` response. So functions like `json` does not exist. 

I tried adding a custom response that included the `json` function, didn't get that to work just yet though.. But not sure if this is the way to.

I did check out https://pypi.org/project/requests-flask-adapter/, which I think looks exactly like something we want:

> FlaskAdapter is a requests adapter intended to allow its user to call Flask app endpoints, with requests, without having to run a Flask server.
> 
> Its main uses include building integration tests and client tests without having to resort to multithreading/multiprocessing/running an instance in another shell/docker to spawn a running Flask app. In doing so, you are able to to call the endpoints you wish to test with your client.

But our current code structure doesn't seem to allow this. Would have to refactor `StorageClient` to accept a `Session` instead - which might be the sensible way to go.

### CODE REVIEW

File: ert_gui/tools/plot/storage_client.py [hunk 1]
```diff
@@ -9,8 +9,8 @@ def convert_response_to_float(resp):
     return [float(x) for x in string.split(",")]
 
 class StorageClient(object):
-    def __init__(self):
-        self._BASE_URI = "http://127.0.0.1:5000/" # Default flask server
+    def __init__(self, base_url=None):
+        self._BASE_URI = base_url or "http://127.0.0.1:5000/" # Default flask server
```
Comment (sondreso) [inline ert_gui/tools/plot/storage_client.py:13]: Not a fan of the implicit bool conversion here :sweat_smile: Would a better idea be to have the default url in the function signature instead? 
`def __init__(self, base_url="http://127.0.0.1:5000/"):`

**Comment by sondreso** [comment]
Think you should coordinate a bit with @oysteoh here, he is working with very related stuff!

**Comment by lars-petter-hauge** [comment]
Yepp, we are discussing this :)

File: ert_gui/tools/plot/storage_client.py (hunk 1 above)
Comment (lars-petter-hauge) [reply ert_gui/tools/plot/storage_client.py:13]: Some like it, some don't I've gotten to experience. Not an issue having it in the signature instead.

**Comment by lars-petter-hauge** [comment]
Issue 1, getting the correct response object returned:


```
    # documentation says response class should be set on test_client
    # that does not have an effect though, and it must be set directly on the wrapper
    # suppose we have a different setup than what is standard.
    flWrapper.app.response_class = CustomResponse
    testing_client = flWrapper.app.test_client()

    # This is from documentation, but does not have an effect in our case
    # testing_client.response_class = CustomResponse
```
  Issue 2, the response object is not subclassed from the same as `requests.models.Response`
```
    # As expected, a lot of functions from the requests.models.Response
    # has to be created. Subclassing requests.models.Response
    # instead, just ends us up in the same inverted issue..
```
So far 3 functions had to be mocked, I expect these to grow..

I suppose it can be as it is for now, and once the HTTP api is more stable, we'll just mock the entire backend

File: ert_gui/tools/plot/storage_client.py (hunk 1 above)
Comment (lars-petter-hauge) [reply ert_gui/tools/plot/storage_client.py]: ps: the reason I didn't add in the signature (which I would have normally of course), is that I though the default value is more of a last resort fallback, or implementation detail. I didn't think it fit within the `StorageClient` at all. Gotta stop over thinking ;)

**Comment by lars-petter-hauge** [comment]
Are you reviewing this @sondreso ?

At least it is ready for review

**Comment by sondreso** [comment]
I was not reviewing it no, just commented something I found when casually looking through :) 
Suggest @oysteoh for review!

**Pull request merged**